pep-0692.rst

```diff
@@ -39,9 +42,66 @@ type annotating ``**kwargs`` is not possible. This is especially a problem for
 already existing codebases where the need of refactoring the code in order to
 introduce proper type annotations may be considered not worth the effort. This
 in turn prevents the project from getting all of the benefits that type hinting
-can provide. As a consequence, there has been a `lot of discussion <mypyIssue4441_>`__
-around supporting more precise ``**kwargs`` typing and it became a
-feature that would be valuable for a large part of the Python community.
+can provide.
+
+Moreover, one pattern that ``**kwargs`` are especially useful for is when a
+function should accommodate optional keyword-only arguments that don't have
+default values. A need for a pattern like that can arise when values that are
+usually used as defaults to indicate no user input, such as ``None``, can be
+passed in by a user and should result in a valid, non-default behavior.
+
+As an example, let's consider a hypothetical ``HTTPClient`` class::
+
+    class HTTPClient:
+        def __init__(
+            self,
+            *,
+            auth: AuthType,
+            timeout: TimeoutType,
+            base_url: UrlType,
+            # And some more arguments...
+        ): ...
+
+        def request(
+            self,
+            *,
+            auth,
+            timeout
+            # And some more arguments...
+        ): ...
+
+        # And a bunch of other methods...
+
+In this case, the default authorization and timeout strategies are passed in
+via the constructor. However, calls to the ``request`` method can override the
+defaults. For example, passing ``auth=None`` explicitly, could mean that no
+authorization strategy should be used at all, and the same could be true for
+``timeout``. In this situation there are no obvious default values for those
+parameters.
+
+This example is based on real world use cases taken from popular `httpx
+<httpxSentinel_>`__ and `urllib3 <urllib3Sentinel_>`__ libraries. Those
+libraries deal with this problem by introducing sentinel objects that are used
+as defaults. One issue of such a solution is that if the clients subclass a
+class that is part of a public API and want to override the ``request`` method,
+they may need to import that sentinel object, which usually should not be made
+public. Using ``**kwargs`` in this example would not have this problem and is
+arguably simpler. However, currently there is no way to properly type hint such
+a solution.
+
+In addition, ``**kwargs`` can be used to reduce the amount of code needed in
+cases when there is a top-level method that is a part of a public API and it
+calls a bunch of helper functions, all of which expect the same keyword
+arguments. Unfortunately, if those helper functions were to use ``**kwargs``,
+there is no way to properly type hint them if the keyword arguments they expect
+are of different types.
```

Comment:
This strikes me as the main use case -- the need to avoid repetition when one function calls another with an identical signature. Maybe you could elaborate more on how frequent this pattern is? I've encountered it enough to think it's quite often -- e.g. I recall writing many wrappers for `open()` that were a pain to write. I've also seen real bugs reported that were due to such a situation where the helper was updated to accept a new keyword and the wrapper wasn't.

(The other legit use case is adding type annotations to a legacy code base that uses `**kwargs` wrappers a lot -- I agree with what you wrote at the top of the Motivation section about exceeding the time budget for the migration.)